tfhe/src/core_crypto/commons/mod.rs

```diff
@@ -77,24 +71,50 @@ pub mod test_tools {
     use crate::core_crypto::commons::traits::*;
     use concrete_csprng::seeders::{Seed, Seeder};
 
-    fn modular_distance<T: UnsignedInteger>(first: T, other: T) -> T {
+    pub fn modular_distance<T: UnsignedInteger>(first: T, other: T) -> T {
         let d0 = first.wrapping_sub(other);
         let d1 = other.wrapping_sub(first);
         d0.min(d1)
     }
 
-    fn torus_modular_distance<T: UnsignedInteger>(first: T, other: T) -> f64 {
-        let d0 = first.wrapping_sub(other);
-        let d1 = other.wrapping_sub(first);
-        if d0 < d1 {
-            let d: f64 = d0.cast_into();
-            d / 2_f64.powi(T::BITS as i32)
+    pub fn torus_modular_distance<T: UnsignedInteger>(
+        first: T,
+        other: T,
+        modulus: CiphertextModulus<T>,
+    ) -> f64 {
+        if modulus.is_compatible_with_native_modulus() {
+            if modulus.is_native_modulus() {
+                let d0 = first.wrapping_sub(other);
+                let d1 = other.wrapping_sub(first);
+                if d0 < d1 {
+                    let d: f64 = d0.cast_into();
+                    d / 2_f64.powi(T::BITS as i32)
+                } else {
+                    let d: f64 = d1.cast_into();
+                    -d / 2_f64.powi(T::BITS as i32)
+                }
+            } else {
+                let modulus = modulus.get_custom_modulus();
+                let d0 = first.wrapping_sub(other);
+                let d1 = other.wrapping_sub(first);
+                if d0 < d1 {
+                    let d: f64 = d0.cast_into();
+                    d / 2_f64.powi(modulus.ilog2() as i32)
+                } else {
+                    let d: f64 = d1.cast_into();
+                    -d / 2_f64.powi(modulus.ilog2() as i32)
+                }
+            }
```

Comment:
Could this become

```rust
let bits = if modulus.is_native_modulus() { 
   T::BITS as i32
} else { 
    modulus.get_custom_modulus().ilog2() as i32
};

let d0 = first.wrapping_sub(other);
let d1 = other.wrapping_sub(first);
 if d0 < d1 {
   let d: f64 = d0.cast_into();
   d / 2_f64.powi(bits)
} else {
   let d: f64 = d1.cast_into();
   -d / 2_f64.powi(bits)
 }
```

Reply:
most likely yes, I was just keeping the original code to not mess things up initially, I'll update